src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php

```diff
@@ -94,43 +111,41 @@ public function getIdentifier()
      */
     public function setModuleName($moduleName)
     {
-        if (
-            null === $this->moduleName || empty($this->moduleName)
-        ) {
+        if (null === $this->moduleName || empty($this->moduleName)) {
             UnsupportedModuleException::moduleNotProvided(self::getIdentifier());
         }
 
         $this->moduleName = $moduleName;
 
+        // ugly hack, I know
+        $this->domain = DomainHelper::buildModuleBaseDomain($moduleName);
+
         return $this;
     }
 
     /**
-     * {@inheritdoc}
+     * @param string $domain
+     *
+     * @return AbstractProvider|SearchProviderInterface|void
      */
-    public function getDefaultCatalogue($empty = true)
+    public function setDomain($domain)
     {
-        $defaultCatalogue = new MessageCatalogue($this->locale);
-
-        try {
-            $defaultCatalogue = $this->moduleProvider
-                ->setModuleName($this->moduleName)
-                ->setLocale($this->locale)
-                ->getDefaultCatalogue()
-            ;
-        } catch (Exception $exception) {
-            // Do nothing if xliff files doesn't exists
-        }
+        throw new InvalidArgumentException(__CLASS__ . ' does not allow calls to setDomain()');
+    }
 
-        try {
-            $additionalDefaultCatalogue = $this->legacyModuleExtractor->extract($this->moduleName, $this->locale);
-            $defaultCatalogue->addCatalogue($additionalDefaultCatalogue);
-        } catch (UnsupportedLocaleException $exception) {
-            // Do nothing as support of legacy file is deprecated
-        }
+    /**
+     * Returns the default (aka untranslated) catalogue
```

Comment:
let inheritdoc please, as this method relies on UseDefaultCatalogueInterface

Reply:
Done